### PR TITLE
Patch openssh handling of smartcard details for CVE-2023-28531

### PR DESCRIPTION
**Pull request by ddstreet**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Patch openssh handling of smartcard details for CVE-2023-28531
###### Change Log  

- Change
- Change
- Change
Patch openssh handling of smartcard details for CVE-2023-28531

###### Does this affect the toolchain?  

NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-28531

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=375177&view=results

### CODE REVIEW

File: SPECS/openssh/openssh.spec [hunk 1]
```diff
@@ -16,6 +16,7 @@ Patch0:         blfs_systemd_fixes.patch
 # Nopatches section
```
Comment (dmcilvaney) [inline SPECS/openssh/openssh.spec:16]: Looks like this was intended for nopatch only, the real patches should be above (`Patch1: CVE-2023-28531-upstream-include-destination-constraints-for-smartca.patch`)

File: SPECS/openssh/openssh.spec [hunk 2]
```diff
@@ -60,6 +61,7 @@ This provides the ssh server daemons, utilities, configuration and service files
 %setup -q
 tar xf %{SOURCE1} --no-same-owner
 %patch0
+%patch101 -p1
```
Comment (dmcilvaney) [inline SPECS/openssh/openssh.spec:64]: Associated with above comment.
```suggestion
%patch1 -p1
```

File: SPECS/openssh/openssh.spec (hunk 1 above)
Comment (ddstreet) [reply SPECS/openssh/openssh.spec:16]: does it matter?

File: SPECS/openssh/openssh.spec (hunk 2 above)
Comment (ddstreet) [reply SPECS/openssh/openssh.spec:64]: same response as above; the patch number doesn't matter...

**Review by hbeberman — APPROVED** [review]

**Pull request merged**